components/script/dom/bindings/buffer_source.rs

```diff
@@ -63,36 +63,25 @@ pub(crate) enum BufferSource {
     ArrayBuffer(Box<Heap<*mut JSObject>>),
 }
 
-pub(crate) fn new_initialized_heap_buffer_source<T>(
-    init: HeapTypedArrayInit,
+pub(crate) fn create_heap_buffer_source_with_length<T>(
+    cx: JSContext,
+    len: u32,
     can_gc: CanGc,
-) -> Result<HeapBufferSource<T>, ()>
+) -> Fallible<HeapBufferSource<T>>
 where
     T: TypedArrayElement + TypedArrayElementCreator,
     T::Element: Clone + Copy,
 {
-    let heap_buffer_source = match init {
-        HeapTypedArrayInit::Buffer(buffer_source) => HeapBufferSource {
-            buffer_source,
-            phantom: PhantomData,
-        },
-        HeapTypedArrayInit::Info { len, cx } => {
-            rooted!(in (*cx) let mut array = ptr::null_mut::<JSObject>());
-            let typed_array_result =
-                create_buffer_source_with_length::<T>(cx, len as usize, array.handle_mut(), can_gc);
-            if typed_array_result.is_err() {
-                return Err(());
-            }
-
-            HeapBufferSource::<T>::new(BufferSource::ArrayBufferView(Heap::boxed(*array.handle())))
-        },
-    };
-    Ok(heap_buffer_source)
-}
+    rooted!(in (*cx) let mut array = ptr::null_mut::<JSObject>());
+    let typed_array_result =
+        create_buffer_source_with_length::<T>(cx, len as usize, array.handle_mut(), can_gc);
+    if typed_array_result.is_err() {
+        return Err(Error::JSFailed);
```

Comment:
Are you sure about the `Error::JSFailed`, previously it was just a `()`. Also the error is mapped to `JSFailed` a thttps://github.com/servo/servo/pull/31398/files#diff-74ee6b1aa263f228fb8648bb921e80b79654506e8b39570b764649a02f881402R61